test/parallel/test-tls-two-cas-one-string.js

```diff
@@ -1,35 +1,50 @@
 'use strict';
 
+const assert = require('assert');
 const common = require('../common');
 const tls = require('tls');
 const fs = require('fs');
 
 const ca1 =
-    fs.readFileSync(`${common.fixturesDir}/keys/ca1-cert.pem`, 'utf8');
+  fs.readFileSync(`${common.fixturesDir}/keys/ca1-cert.pem`, 'utf8');
 const ca2 =
-    fs.readFileSync(`${common.fixturesDir}/keys/ca2-cert.pem`, 'utf8');
+  fs.readFileSync(`${common.fixturesDir}/keys/ca2-cert.pem`, 'utf8');
 const cert =
-    fs.readFileSync(`${common.fixturesDir}/keys/agent3-cert.pem`, 'utf8');
+  fs.readFileSync(`${common.fixturesDir}/keys/agent3-cert.pem`, 'utf8');
 const key =
-    fs.readFileSync(`${common.fixturesDir}/keys/agent3-key.pem`, 'utf8');
+  fs.readFileSync(`${common.fixturesDir}/keys/agent3-key.pem`, 'utf8');
 
-function test(ca, next) {
-  const server = tls.createServer({ ca, cert, key }, function(conn) {
-    this.close();
-    conn.end();
-  });
 
-  server.addContext('agent3', { ca, cert, key });
+function test(ca) {
+  return new Promise(function(resolve, reject) {
```

Comment:
I would definitely switch to _promises_ here; since, now there are _six_ tests compared to earlier _two_. 

This approach is already been used in other tests, for e.g., [here](https://github.com/nodejs/node/blob/e84c69310f742c1b2381fa19b155db6f178af7da/test/parallel/test-net-server-max-connections-close-makes-more-available.js#L87). If that helps with the review process